packages/tupaia-web/src/features/Dashboard/DashboardMenu.tsx

```diff
@@ -73,15 +74,78 @@ const DashboardMenuItem = ({ dashboardName, onClose }: DashboardMenuItemProps) =
   );
 };
 
+interface ActionsMenuProps {
+  setExportModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const ActionsMenu = ({ setExportModalOpen}: ActionsMenuProps) => {
+  const [isSubscribed, setIsSubscribed] = useState<boolean>(false)
+  const [mailingListEnabled, setMailingListEnabled] = useState<boolean>(false)
+  const { projectCode, entityCode, dashboardName } = useParams();
+  const {
+    activeDashboard,
+  } = useDashboards(projectCode, entityCode, dashboardName);
+  
+
+
+  const handleSubscribe = (isUserSubscribed) => {
+    // TODO: Add subscribe and unsubscribe sequence
+    // Add react query to POST request for add or remove subscription
+    // Optional - add loading indicator to option.
+    console.log(isUserSubscribed)
+  } 
+  useEffect(() => {
+    const { mailingLists } = activeDashboard;
+    const { mailingListEntity, userSubscribed } = mailingLists.find(({mailingListEntityCode}) => {
+    mailingListEntityCode === entityCode
+    })
+    if(!mailingListEntity) {
+      setMailingListEnabled(false)
+      setIsSubscribed(false)
+    }
+    if(!userSubscribed) {
+      setMailingListEnabled(true)
+      setIsSubscribed(false)
+    }
+    setIsSubscribed(true)
+  },[activeDashboard])
```

Comment:
I think this might an unnecessary use of `useEffect` here. We're just deriving some local variables from the `activeDashboard` here, so I think we don't need to store them in state hooks like this.
```suggestion
  const { projectCode, entityCode, dashboardName } = useParams();
  const {
    activeDashboard,
  } = useDashboards(projectCode, entityCode, dashboardName);
  
  const mailingList = activeDashboard?.mailingLists
    ? activeDashboard.mailingLists.find(({ entityCode: mailingListEntityCode }) => {
        mailingListEntityCode === entityCode;
      })
    : undefined;

  const handleSubscribe = (isUserSubscribed) => {
    // TODO: Add subscribe and unsubscribe sequence
    // Add react query to POST request for add or remove subscription
    // Optional - add loading indicator to option.
    console.log(isUserSubscribed)
  } 
```